src/Microsoft.Diagnostics.Runtime/src/DataReaders/IMemoryReader.cs

```diff
@@ -41,6 +41,16 @@ public interface IMemoryReader
         /// <returns>The value at addr, or default(T) if not present in the data target.</returns>
         T Read<T>(ulong address) where T : unmanaged;
 
+        /// <summary>
+        /// Reads <c>values.Length</c> unmanaged values from the given address into <paramref name="values"/>.
+        /// </summary>
+        /// <typeparam name="T">The type to read.  This may be a struct or primitive type as long as it does
+        /// not managed pointeres.</typeparam>
+        /// <param name="address">The address to read from.</param>
+        /// <param name="values">An array the read values will be written into.</param>
+        /// <returns><see langword="true"/> if <c>values.Length</c> were read; else <see langword="false"/>.</returns>
+        bool ReadArray<T>(ulong address, T[] values) where T : unmanaged;
```

Comment:
I don't want to add this to IMemoryReader.  We can accomplish this with just reading `Span<byte>` of the right size.

I try to keep `IMemoryReader` and `IDataReader` as simple as possible (within reason) so that we don't have to implement a lot across a lot of implementations.

Instead it would make sense to just do the `MemoryMarshal.Cast` on the caller site wherever its needed.

Reply:
Good idea, when I first wrote it, I didn't actually really know the IMemoryReader interface, so I was just writing the code down there.
Good observation that it can be accomplished without adding new methods.